qa/tasks/cephfs/test_nfs.py

```diff
@@ -647,15 +647,26 @@ def test_cluster_set_user_config_with_non_existing_clusterid(self):
         '''
         Test setting user config for non-existing nfs cluster.
         '''
-        try:
-            cluster_id = 'invalidtest'
-            self.ctx.cluster.run(args=['ceph', 'nfs', 'cluster',
-                'config', 'set', self.cluster_id, '-i', '-'], stdin='testing')
-            self.fail(f"User config set for non-existing cluster {cluster_id}")
-        except CommandFailedError as e:
-            # Command should fail for test to pass
-            if e.exitstatus != errno.ENOENT:
-                raise
+        cluster_id = 'invalidtest'
+        with contextutil.safe_while(sleep=3, tries=3) as proceed:
+            while proceed():
+                try:
+                    self.ctx.cluster.run(args=['ceph', 'nfs', 'cluster',
+                                               'config', 'set', cluster_id,
+                                               '-i', '-'], stdin='testing')
+                    self.fail(f"User config set for non-existing cluster"
+                              f"{cluster_id}")
+                except contextutil.MaxWhileTries:
+                    log.error('error encountered before timing out: '
+                              f'{last_error}')
+                    raise
+                except CommandFailedError as e:
+                    # Command should fail for test to pass
+                    if e.exitstatus == errno.ENOENT:
+                        break
+                    log.warning('exitstatus != 2, retrying')
+                    last_error = str(e)
```

Comment:
ditto: last_error required?

Reply:
can be helpful to identify what caused the failure while debugging

Reply:
i'll remove this

Reply:
removed